### PR TITLE
[skip ci] Add new docs

### PR DESCRIPTION
**Pull request by PathogenDE**

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -1,16 +1,59 @@
...
+## Structure
 
-In here you will find all the information you need to contribute, 
-as well as the architecture, and understanding how the Operator works.
+* Folder ./architecture Information about the software architecture and inner working of the operator
+* Folder ./ops-guide: Information about the operations of the operator
+* Folder ./developer-guide: Information how to interact with the operator as a developer
+* [Goals page](goals.md) Goals/Objectives the Operator tries to accomplish
+* [Limitations](limitations.md) Current limitation the operator is operating under
+* [Roadmap](TOADMAP.md) Future roadmap of the operator. Where we want to go
```
Comment (andreas-kr) [inline docs/README.md:10]: [Roadmap](TOADMAP.md) unless it's a map about toads, probably the link is wrong :D

File: docs/README.md [hunk 2]
```diff
@@ -1,16 +1,59 @@
...
+
+* The Operator will not scale an application if the application is not opted in. 
+
+* The Operator will not scale if there's no scalingstate present. (Cluster-wide or namespace-wide)
+
+* **Scaling an aplication:** If a scalingstate is active, the application is opted in, and the application has that scalingstate defined, the operator will scale the deployment to the specified replica count as per the scaling state.
+
+* **Modification of `spec.replicas`:** If an application is opted in, and the application replica count is manipulated from elsewhere, the operator will set the replica count back to the replica count of the specific active scaling state.
+
+* **Modification of a state:** If there's a change to the replica-count for a state, the operator will take that change into account as long as the application is opted in. There is also an optional `admission-controller` that can prevent that modification all-together
```
Comment (andreas-kr) [inline docs/README.md:49]: Let's add the repo link in the admission controller maybe?

File: docs/developer-guide/developer-guide.md [hunk 3]
```diff
@@ -29,6 +29,8 @@ spec:
     description: "Maximum scale settings."
     priority: 1
     rate-limit: {}
```
Comment (andreas-kr) [inline docs/developer-guide/developer-guide.md:31]: I think we can remove this now right

File: docs/README.md (hunk 1 above)
Comment (andreas131989) [inline docs/README.md:10]: Unless it's a map about toads, probably the link is wrong :D

File: docs/README.md (hunk 2 above)
Comment (andreas131989) [inline docs/README.md:49]: Let's add the repo link in the admission controller maybe?

File: docs/developer-guide/developer-guide.md (hunk 3 above)
Comment (andreas131989) [inline docs/developer-guide/developer-guide.md:31]: I think we can remove the rate-limit from here now right

Comment (PathogenDE) [reply docs/developer-guide/developer-guide.md:31]: yes

File: docs/README.md (hunk 2 above)
Comment (PathogenDE) [reply docs/README.md:49]: Yup, good point!

File: docs/README.md (hunk 1 above)
Comment (PathogenDE) [reply docs/README.md:10]: No documentation about Toad - yet. We have Roadmaps though :P

**Review by andreas131989 — APPROVED** [review]

**Pull request merged**